test/common/config/http_subscription_impl_test.cc

```diff
@@ -42,6 +42,26 @@ TEST_F(HttpSubscriptionImplTest, BadJsonRecovery) {
   verifyStats(3, 1, 0, 1, 7148434200721666028);
 }
 
+TEST_F(HttpSubscriptionImplTest, ConfigNotModified) {
+  startSubscription({"cluster0", "cluster1"});
+  EXPECT_CALL(random_gen_, random()).WillOnce(Return(0));
+  EXPECT_CALL(*timer_, enableTimer(_));
+  EXPECT_CALL(callbacks_, onConfigUpdateFailed(_));
+  http_callbacks_->onFailure(Http::AsyncClient::FailureReason::Reset);
```

Comment:
Why do we start with `onFailure`?

Reply:
No reason personally, happy to change it. I don't really know what it does 😅

Reply:
Sounds good, I think we can remove this line and any expects above that were related.